### PR TITLE
[KEYCLOAK-12161] "Back to Application" link is shown with link to current page

### PR DESCRIPTION
**Pull request by douglaspalmer**

### CODE REVIEW

**Comment by vmuzikar** [comment]
@douglaspalmer Please let me know when this is ready for final review.

**Comment by stianst** [comment]
Remember that existing clients may already have an empty value for the baseUrl - so the correct fix at least for now is to check if the baseUrl is empty. In the future we should really not set the value in the first place if it's empty, but that's something that would need a pretty big update to admin console since empty values in a field is sent as an empty string by Angular. We can probably address that in admin console v2 though.

**Comment by douglaspalmer** [comment]
> @douglaspalmer Please let me know when this is ready for final review.



> Remember that existing clients may already have an empty value for the baseUrl - so the correct fix at least for now is to check if the baseUrl is empty. In the future we should really not set the value in the first place if it's empty, but that's something that would need a pretty big update to admin console since empty values in a field is sent as an empty string by Angular. We can probably address that in admin console v2 though.

I had that discussion with @vmuzikar yesterday and updated the PR to reflect that

File: testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/forms/MultipleTabsLoginTest.java [hunk 1]
```diff
@@ -368,4 +373,53 @@ public void loginWithSameClientDifferentStatesLoginInTab2() throws Exception {
...
+    @Test
+    public void testEmptyBaseUrl() throws Exception {
+        String clientUuid = KeycloakModelUtils.generateId();
+        ClientRepresentation emptyBaseclient = ClientBuilder.create()
+                .clientId("empty-baseurl-client")
+                .id(clientUuid)
+                .enabled(true)
+                .baseUrl("")
+                .addRedirectUri("http://localhost:8180/auth/realms/master/app/auth/*")
+                .addRedirectUri("https://localhost:8543/auth/realms/master/app/auth/*")
```
Comment (vmuzikar) [inline testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/forms/MultipleTabsLoginTest.java]: Just a small nitpick. I think we could just use single `.addRedirectUri("*")`.

**Review by vmuzikar — APPROVED** [review]
Found one insignificant nitpick, approving regardless – LGTM. Thanks @douglaspalmer!

**Review by vmuzikar — CHANGES_REQUESTED** [review]
I was too fast. ;) Please change the commit message and PR name. We're not setting baseURL to null instead of empty string.

File: testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/forms/MultipleTabsLoginTest.java (hunk 1 above)
Comment (douglaspalmer) [reply testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/forms/MultipleTabsLoginTest.java]: Ok, I'll fix that.

**Comment by douglaspalmer** [comment]
> I was too fast. ;) Please change the commit message and PR name. We're not setting baseURL to null instead of empty string.

I'll fix that too.

**Comment by abstractj** [comment]
@douglaspalmer out of curiosity: why not change to the title of the same Jira https://issues.redhat.com/browse/KEYCLOAK-12161 ? Or maybe we should change the Jira?

**Comment by vmuzikar** [comment]
@abstractj The JIRA summary (title) is fine, it describes the bug: ""Back to Application" link is shown with link to current page"

**Review by vmuzikar — APPROVED** [review]
LGTM now, thanks!

**Review by abstractj — APPROVED** [review]

**Pull request merged**